lms/djangoapps/grades/tests/test_services.py

```diff
@@ -0,0 +1,239 @@
+        }
+
+    def test_get_subsection_grade(self):
+        self.assertDictEqual(self.subsection_grade_to_dict(self.service.get_subsection_grade(
+            user_id=self.user.id,
+            course_key_or_id=self.course.id,
+            usage_key_or_id=self.subsection.location
+        )), {
+            'earned_all': 6.0,
+            'earned_graded': 5.0
+        })
+
+        # test with id strings as parameters instead
+        self.assertDictEqual(self.subsection_grade_to_dict(self.service.get_subsection_grade(
+            user_id=self.user.id,
+            course_key_or_id=unicode(self.course.id),
+            usage_key_or_id=unicode(self.subsection.location)
+        )), {
+            'earned_all': 6.0,
+            'earned_graded': 5.0
+        })
+
+    def test_get_subsection_grade_override(self):
+        override, _ = PersistentSubsectionGradeOverride.objects.update_or_create(grade=self.grade)
+
+        self.assertDictEqual(self.subsection_grade_override_to_dict(self.service.get_subsection_grade_override(
+            user_id=self.user.id,
+            course_key_or_id=self.course.id,
+            usage_key_or_id=self.subsection.location
+        )), {
+            'earned_all_override': override.earned_all_override,
+            'earned_graded_override': override.earned_graded_override
+        })
+
+        override, _ = PersistentSubsectionGradeOverride.objects.update_or_create(
+            grade=self.grade,
+            defaults={
+                'earned_all_override': 9.0
+            }
+        )
+
+        # test with id strings as parameters instead
+        self.assertDictEqual(self.subsection_grade_override_to_dict(self.service.get_subsection_grade_override(
+            user_id=self.user.id,
+            course_key_or_id=unicode(self.course.id),
+            usage_key_or_id=unicode(self.subsection.location)
+        )), {
+            'earned_all_override': override.earned_all_override,
+            'earned_graded_override': override.earned_graded_override
+        })
+
+    @ddt.data(
+        [{
+            'earned_all': 0.0,
+            'earned_graded': 0.0
+        }, {
+            'earned_all': 0.0,
+            'earned_graded': 0.0
+        }],
+        [{
+            'earned_all': 0.0,
+            'earned_graded': None
+        }, {
+            'earned_all': 0.0,
+            'earned_graded': 5.0
+        }],
+        [{
+            'earned_all': None,
+            'earned_graded': None
+        }, {
+            'earned_all': 6.0,
+            'earned_graded': 5.0
+        }],
+        [{
+            'earned_all': 3.0,
+            'earned_graded': 2.0
+        }, {
+            'earned_all': 3.0,
+            'earned_graded': 2.0
+        }],
+    )
+    @ddt.unpack
+    def test_override_subsection_grade(self, override, expected):
+        PersistentSubsectionGradeOverride.objects.all().delete()  # clear out all previous overrides
```

Comment:
I'd argue this is better served in the class `tearDown` method, so we can ensure a clean state for every test.